src/providers/spatialite/qgsspatialiteprovider.cpp

```diff
@@ -529,7 +532,7 @@ QgsSpatiaLiteProvider::QgsSpatiaLiteProvider( QString const &uri )
     return;
   }
   mEnabledCapabilities = mPrimaryKey.isEmpty() ? QgsVectorDataProvider::Capabilities() : ( QgsVectorDataProvider::SelectAtId );
-  if ( ( mTableBased || mViewBased ) &&  !mReadOnly )
+  if ( mTableBased )
```

Comment:
The ReadOnly applies only to views in Spatialite.
Here, I believe, it is only used together with views or VirtualShapes (which are always readonly).
There are no triggers for TABLES that make them writable.